metal/devices.go

```diff
@@ -188,6 +193,66 @@ func (i *instances) InstanceShutdownByProviderID(_ context.Context, providerID s
 	return device.State == "inactive", nil
 }
 
+// InstanceShutdown returns true if the node is shutdown in cloudprovider
+func (i *instances) InstanceShutdown(ctx context.Context, node *v1.Node) (bool, error) {
+	return i.InstanceShutdownByProviderID(ctx, node.Spec.ProviderID)
+}
+
+// InstanceExists returns true if the node exists in cloudprovider
+func (i *instances) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
+	return i.InstanceExistsByProviderID(ctx, node.Spec.ProviderID)
```

Comment:
I really would like to have this be more flexible, using your new `deviceByNode()`, so it can handle with or without a provider ID. But as [InstancesV2](https://pkg.go.dev/k8s.io/cloud-provider#InstancesV2) says that `InstanceExists` and `InstanceShutdown` could use either, let's leave this. I will open a new PR to do that once this is in.